CONTRIBUTING.md

```diff
@@ -1,219 +1,193 @@
+
+### Setting up your development environment
+
+#### Setup IntelliJ IDEA
+
+##### Importing a Ballerina Project
+
+Import the Ballerina project to IntelliJ IDEA similar to any other gradle project. 
+
+* Navigate to `File | Open` menu.
+* Browse the filesystem and Select Ballerina Project's root directory.
+* In "Import Project from Maven" Wizard, set followings.
+   * If prompted to import gradle project select `Auto import`.
+   * Set Project SDK as Java 1.8
+
+##### Useful IDEA Plugins
+
+* ANTLR 4 - https://plugins.jetbrains.com/plugin/7358-antlr-v4-grammar-plugin 
+* Ballerina - https://plugins.jetbrains.com/plugin/9520-ballerina (Need IDEA 2016.3 or newer)
+
+#### Setup Eclipse
+
+##### Importing a Ballerina Project
+
+All Ballerina repositories are developed as maven or gradle projects. So you can import any Ballerina project to Eclipse similar to any maven or gradle project. 
+
+* Navigate to **File** → **Import..**
+* Select **Existing Maven Projects** under **Maven**
+* Next, browse the file system, and open the Maven module.
+
+##### Installing ANTLR4 plugin
+If you are working with the grammar, it would be useful to have ANTLR 4 IDE plugin installed for Eclipse. To install this using Eclipse marketplace:
+
+1. Navigate to **Help** → **Eclipse Marketplace**
+
+2. Search for **antlr4**
+
+3. Install ANTLR 4 IDE plugin (which supports antlr 4.x version).
+
+To open the plugin views: 
+
+1. Navigate to **Window** → **Show View** → **Other**. 
+
+2. Under **ANTLR4** section, select and enable **Parse Tree** view and **Syntax Diagram** views.
+
+### Working With Ballerina Grammar
+
+Ballerina grammar has been implemented using ANTLR4. It is recommended to have some basic understanding on anltr4 grammar syntax and concepts, before working with the Ballerina grammar. “Parr, Terence (January 15, 2013), The Definitive ANTLR 4 Reference” is a good reference to get started with.
+
+Ballerina grammar can be found under docs/grammar in the Ballerina repository. It consists of two files:
+
+* BallerinaLexer.g4 - Contains the lexer rules for Ballerina grammar. Lexer is responsible for tokenizing an input Ballerina source code.
+* BallerinaParser.g4 - Contains the parser rules. Parser listens to the token stream generated by the lexer. High level grammar productions/abstractions are defined in the parser using those tokens.
+
+> TIP: If you want to check and validate a grammar rule you just wrote, you can use the antlr4 plugin of your IDE. See Setting up your development environment section for more details.
+
+#### Generating Parsers
+
+Once a change is done to the any of the grammar files, it is needed to re-generate the parsers. To generate parsers, we first need to download `antlr-complete-4.5.3.jar` file. Then navigate to `<ballerina>/compiler/ballerina-lang/src/main/resources/grammar/`, and generate the lexer followed by the parser (order is important) using the following command:
```

Comment:
```suggestion
Once a change is done to any of the grammar files, the lexer and the parser need to be re-generated. To generate the lexer and the parser, first download the `antlr-complete-4.5.3.jar` file. Then navigate to `<ballerina>/compiler/ballerina-lang/src/main/resources/grammar/`, and execute the following command:
```
Removed that bit about generating the lexer before parser because when we run the command as mentioned, we don't have to worry about an order.